src/WebJobs.Script.Abstractions/Description/Binding/BindingMetadata.cs

```diff
@@ -95,6 +108,11 @@ public static BindingMetadata Create(JObject raw)
             bindingMetadata.Connection = connection;
             bindingMetadata.Raw = raw;
 
+            if (raw.TryGetValue("properties", StringComparison.OrdinalIgnoreCase, out JToken value))
+            {
+                bindingMetadata.Properties = new Dictionary<string, object>(value.ToObject<IDictionary<string, object>>(), StringComparer.OrdinalIgnoreCase);
```

Comment:
Instead of assigning a new dictionary entirely, can you add the range of values from the `JToken` to the existing dictionary? Then the `private set` can be removed.

Alternatively, add a private ctor which takes in `Dictionary<>` and call that here:

``` CSharp
public BindingMetadata() : this(new Dictionary<string, object>()) { }

private BindingMetadata(Dictionary<string, object> properties)
{
    Properties = properties;
}

public static BindingMetadata Create(JObject raw)
{
  // refactor to pull out `properties` and use in ctor.
}
```